api/test/integration/api-key-guard.e2e-spec.ts

```diff
@@ -0,0 +1,45 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import request from 'supertest';
+import { AppModule } from '../../src/modules/app.module';
+import { PrismaService } from '../../src/services/prisma.service';
+
+describe('API Key Guard Tests', () => {
+  let app: INestApplication;
+  let prisma: PrismaService;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await prisma.$disconnect();
+    await app.close();
+  });
+
+  it('should succeed when correct header is present', async () => {
+    await request(app.getHttpServer())
+      .get('/jurisdictions')
+      .set({ passkey: process.env.API_PASS_KEY || '' })
```

Comment:
Will the `process.env` cause an issue in doorway tests?

Reply:
it shouldn't, we only run into testing issues on unit tests that attempt to hit the db

this is an e2e test so the db will be present, and the api is stood up in its entirety to test e2e